.gitmodules

```diff
@@ -1,3 +1,4 @@
 [submodule "microsoft-authentication-library-common-for-android"]
 	path = common
 	url = https://github.com/AzureAD/microsoft-authentication-library-common-for-android.git
+	branch = dev
```

Comment:
@iambmelt @heidijinxujia do we really want to do this?

Reply:
If we leave the branch empty, by default, it will fetch the master branch of common-core which is an empty code base now. Without the branch config, the developer who first clones the MSAL repo will encounter the error on finding the :common project. 

Thus, we need to config the branch to dev in the .gitmodules.

Reply:
@heidijinxujia This isn't correct -- the submodule will fetch whichever commit the submodule pointed to when the last commit was made _containing_ the submodule.

The SHA doesn't need to be from a specific branch, it only needs to exist in the source tree. Usually you see this indicated by an `@ <sha>` in the filename

See below:
![capture](https://user-images.githubusercontent.com/990063/39549357-8cdb002a-4e2b-11e8-83b6-66e822bc6d47.JPG)

All this to say, let's **not** explicitly point the submodule reference to a branch as it will make feature branches cumbersome and potentially even _more_ confusing

Reply:
Sorry, I probably I have a different thought.

This is a more robust way to update the working tree of the submodules and get rid of fetching the submodule change in every PR. If we point the gitmodules to a branch, for example 'dev', we will not need to commit the commin@SHA everytime. Instead, git gives us a way to fetch the change automatically by calling
`git submodule init`
`git submodule update --remote`

Well, it quite depends the usage of the submodule. As the common repo is not released yet, I would like to leave the decision to you and Shane.

Reply:
My preference would be to _not_ specify a branch and to instead let it point to a commit SHA. My rationale is that this makes it easier for feature branches to point to other branches/commits while in developments and also that it reduces the likelihood of something changing the behavior or api of a submodule depedency 'beneath' this repo. Example: Someone modifies the API of class in `common` without updating calls to it from `MSAL` (or elsewhere).

@shoatman ^^ Take a look at the above and weigh-in